src/main/webapp/components/NoteGridCard.vue

```diff
@@ -42,12 +69,20 @@
     methods: {
       onClick: function() {
         this.$emit('click');
+      },
+      toggleFavorite: function() {
+        let method = this.favorited ? 'DELETE' : 'POST';
+        this.favorited = !this.favorited;
+        fetch('/favorite-note?note_id=' + this.id, {
+          method: method
+        })
```

Comment:
I think this toggleFavorite method can be deleted because the toggling function should be in NotePreview instead.

Reply:
Good catch